flink-autoscaler/src/main/java/org/apache/flink/autoscaler/event/AutoScalerEventHandler.java

```diff
@@ -33,19 +35,15 @@
 @Experimental
 public interface AutoScalerEventHandler<KEY, Context extends JobAutoScalerContext<KEY>> {
 
-    /**
-     * Handle the event.
-     *
-     * @param interval When interval is great than 0, events that repeat within the interval will be
-     *     ignored.
-     */
+    /** Handle the event. */
```

Comment:
I think the entire eventHandler interface is getting pretty weird. It was already not very nice before this change but now it's even worse because we are leaking too many internals of the operator event triggering mechanism. 

I think we need to break up the handleEvent into two new methods.

```
// Would be the current logic but without the interval logic
handleGenericAutoscalerEvent(...)

// Handle scaling events separately. Then the operator can hide the complex interval/label triggering logic
handleScalingEvent(Context,  Map<JobVertexID, ScalingSummary> scalingSummaries, scalingEnabled)
```

Reply:
And we could even have a default implementation within the interface itself to turn it into a generic event in case we are not running on the operator

Reply:
Thank you for calling out. I was not sure if I should have a separate method, not sure what this handler interfaceshould be handing. We can move the logic of handleGenericAutoscalerEven to the interface and make it an abstract class. What do you think? @gyfora

Reply:
I don't understand the suggestion @clarax what do you think my proposal? Have those 2 methods in the interface and add a default implementation for `handleScalingEvent`.

Reply:
Updated for two methods.  I don't understand "a default implementation within the interface itself to turn it into a generic event in case we are not running on the operator". I made the second method a generic method for any suppression callers want to use.

Reply:
The reason for removing the interval from this interface is because interval just works for kubernetesEventHandler, right? 

I wanna whether it's needed for Standalone Autoscaler, if it's needed, we should keep the `interval` here.

Actually, I prefer keep it first. And removing it if we are sure it's really not needed in the future. WDYT? cc @gyfora 

Also, if the interval is not needed here. `AutoScalerOptions#SCALING_REPORT_INTERVAL` should be moved into `flink-kubernetes-operator` module.

Reply:
I agree, we can keep the interval as a fixed parameter to the reporter methods. That way we don't need custom config handling in the operator.

Reply:
Updated the PR with the change. Added comments to explain too.